### PR TITLE
Up with the team fork

### PR DESCRIPTION
**Pull request by sgdc3**

### CODE REVIEW

**Comment by sgdc3** [comment]
Wait, i have to fix one thing before merging

File: src/main/java/fr/xephi/authme/listener/AuthMePlayerListener.java [hunk 1]
```diff
@@ -428,13 +428,21 @@ public void onPlayerLogin(PlayerLoginEvent event) {
...
             if (!plugin.database.isAuthAvailable(name)) {
                 event.setKickMessage(m.send("reg_only")[0]);
                 event.setResult(PlayerLoginEvent.Result.KICK_OTHER);
                 return;
             }
         }
+        
+        if (Settings.antiBotInAction){
+            if (!plugin.database.isAuthAvailable(name)) {
+                event.setKickMessage(m.send("AntiBot service in action! Non registered players can't connect until the bot attack stops!")[0]); //Need to add string to messages
```
Comment (Xephi) [inline src/main/java/fr/xephi/authme/listener/AuthMePlayerListener.java]: you will get a null pointer by calling this ^^", cause that string doesn't exist

File: src/main/java/fr/xephi/authme/process/register/ProcessSyncronousEmailRegister.java [hunk 2]
```diff
@@ -28,21 +29,26 @@ public ProcessSyncronousEmailRegister(Player player, AuthMe plugin) {
...
         int time = Settings.getRegistrationTimeout * 20;
         int msgInterval = Settings.getWarnMessageInterval;
-        if (time != 0) {
-            LimboCache.getInstance().getLimboPlayer(name).getTimeoutTaskId().cancel();
-            BukkitTask id = Bukkit.getScheduler().runTaskLaterAsynchronously(plugin, new TimeoutTask(plugin, name, player), time);
-            LimboCache.getInstance().getLimboPlayer(name).setTimeoutTaskId(id);
+        
+        BukkitScheduler sched = plugin.getServer().getScheduler();
+        if (time != 0 && limbo != null) {
+        	limbo.getTimeoutTaskId().cancel();
```
Comment (Xephi) [inline src/main/java/fr/xephi/authme/process/register/ProcessSyncronousEmailRegister.java]: wtf with indentation? x)

**Comment by sgdc3** [comment]
I know I have to revert lot's of thing :)

File: src/main/java/fr/xephi/authme/AuthMe.java [hunk 3]
```diff
@@ -277,9 +279,9 @@ public void run() {
                 }
             } catch (Exception ex) {
             }
+        }
 
-        if (Settings.usePurge)
```
Comment (Xephi) [inline src/main/java/fr/xephi/authme/AuthMe.java]: Nop ! let user the choice !

Comment (sgdc3) [inline src/main/java/fr/xephi/authme/AuthMe.java]: Nope it was a duble check, look into the autoPurge method ;)

**Pull request merged**